contracts/DVD/DVDTransferManager.sol

```diff
@@ -250,12 +264,14 @@ contract DVDTransferManager is Ownable {
         address _counterpart,
         address _token2,
         uint256 _token2Amount) external {
-        require(IERC20(_token1).balanceOf(msg.sender) >= _token1Amount, "Not enough tokens in balance");
-        require(
-            IERC20(_token1).allowance(msg.sender, address(this)) >= _token1Amount
-            , "not enough allowance to initiate transfer");
-        require (_counterpart != address(0), "counterpart cannot be null");
-        require(IERC20(_token2).totalSupply() != 0, "invalid address : address is not an ERC20");
+        uint256 balance = IERC20(_token1).balanceOf(msg.sender);
+        require(balance >= _token1Amount, CommonErrors.ERC20InsufficientBalance(msg.sender, balance, _token1Amount));
+
+        uint256 allowance = IERC20(_token1).allowance(msg.sender, address(this));
+        require(allowance >= _token1Amount, CommonErrors.ERC20InsufficientAllowance(msg.sender, allowance, _token1Amount));
+
+        require (_counterpart != address(0), InvalidArgumentErrors.ZeroAddress());
+        require(IERC20(_token2).totalSupply() != 0, InvalidArgumentErrors.AddressNotERC20(_token2));
```

Comment:
in facts i am not sure this error can throw. In case the address is not an ERC20 the call to `totalSupply()` will revert, not return 0 i think. In one way it is good as it reverts, as expected, but it will not revert with the error we want. Can solve this with a `try/catch` block i guess